0002/README.md

```diff
@@ -0,0 +1,167 @@
+---
+id: 0002
+title: Set the mood around releasing code
+status: ideation
+authors: Gianluca Arbezzano <gianarb92@gmail.com>
+---
+
+## Summary
+
+This proposal sets the expectation about what it means to release code part of
+the Tinkerbell organisation.
+
+## Goals and not Goals
+
+Goals are:
+
+1. What is a release
+2. Set a few rules that all the projects should follow
+3. Quality assertion.
+
+Not-Goals:
+
+1. How each project should be released
+2. Define who own releases
+3. Distribution via `apt`, `apk` and so on.
+
+## Content
+
+A release is a tested and stable point in time snapshot from a codebase.
+
+We follow the [Semantic Versioning (semver) 2.0.0](https://semver.org).
+Summarized as follows:
+
+Given a version number MAJOR.MINOR.PATCH, increment the:
+
+* MAJOR version when you make incompatible API changes,
+* MINOR version when you add functionality in a backwards compatible manner, and
+* PATCH version when you make backwards compatible bug fixes.
+
+Additional labels for pre-release and build metadata are available as extensions
+to the MAJOR.MINOR.PATCH format.
+
+### Schedule
+
+Every subproject should be able to follow its own cadence in terms of how often
+or regularly a new release should be cut, as long as it is documented and
+shared with the community.
+
+A new `MINOR` version for Tinkerbell should be released following a proper
+schedule, right now I think every two months sounds reasonable looking at the
+code written and merged in the last couple of weeks. In this way the
+subprojects will have an intrinsic release target if they want to get their new
+code included in the Tinkerbell release.
+
+`PATCH` release for Tinkerbell can be done as required by the subprojects
+bug fixes.
+
+`MAJOR` release right now are not even evaluated because the project is too young.
```

Comment:
This is fair. If we had a roadmap we might populate it with the features that are needed to reach 1.0.0.
https://github.com/tinkerbell/tink/pull/241 😈

Reply:
Major should be described before Minor, to match the earlier bullets.

Reply:
+1 to linking to an umbrella issue, milestone, or some other way of tracking issues needed to reach 1.0

Reply:
ROADMAP is out of scope from this proposal but I am happy to read one about it

Reply:
fwiw, I **really** think we should stamp 0.1.0 where we are and worry about where we are going later. We need a relative starting point (tinkerbell/tink#74). 

0.1.0 is preferred to 0.0.1, https://semver.org/#how-should-i-deal-with-revisions-in-the-0yz-initial-development-phase

Reply:
This proposal is about how we should do releases. Asking to tag 0.1.0 where we are looks unreasonable to me. We don't know where we are and we are not sure about where to go at this phase. The coordination required to get this kind of information for all the projects looks out of scope and not that easy.

Even if we tag a release today we don't have the right setup to use that in docker-compose or in other places and the binaries won't be generated. So I think it is a waste of time and I prefer to make it better.

Reply:
I agree with @displague that we need to get something out there that is "stable" enough for the end-user, as they need something to get their hands on that "just-works™". As development continues on Tinkerbell we've broken various components multiple times and left end-users with no way to know what they're running or anyway of rolling-back. We're looking at pinning a set of `image:sha` until all of the micro services have maintainers and releases.

Reply:
btw I agree with you as well, I just think it is out of scope for this proposal :) now we have #11 that we can use to discuss that and I am happy enough if @displague and you (@thebsdbox) are happy as well!

Reply:
I continued my rant in https://github.com/tinkerbell/proposals/pull/11#pullrequestreview-472114109 and then came back to this PR after another review. 

I'm looking forward to getting this merged soon. I've made some recommendations to pull more topics out-of-scope to assist that.